src/api/providers/vscode-lm.ts

```diff
@@ -258,71 +258,12 @@ export class VsCodeLmHandler implements ApiHandler, SingleCompletionHandler {
 			// Use 4 character-to-token ratio for Claude models
 			return Math.ceil(textContent.length / 4)
 		}
-
-		// Check for required dependencies
-		if (!this.client) {
-			console.warn("Cline <Language Model API>: No client available for token counting")
-			return 0
-		}
-
-		if (!this.currentRequestCancellation) {
-			console.warn("Cline <Language Model API>: No cancellation token available for token counting")
-			return 0
-		}
-
-		// Validate input
-		if (!text) {
-			console.debug("Cline <Language Model API>: Empty text provided for token counting")
-			return 0
-		}
-
-		try {
-			// Handle different input types
-			let tokenCount: number
-
-			if (typeof text === "string") {
-				tokenCount = await this.client.countTokens(text, this.currentRequestCancellation.token)
-			} else if (text instanceof vscode.LanguageModelChatMessage) {
-				// For chat messages, ensure we have content
-				if (!text.content || (Array.isArray(text.content) && text.content.length === 0)) {
-					console.debug("Cline <Language Model API>: Empty chat message content")
-					return 0
-				}
-				tokenCount = await this.client.countTokens(text, this.currentRequestCancellation.token)
-			} else {
-				console.warn("Cline <Language Model API>: Invalid input type for token counting")
-				return 0
-			}
-
-			// Validate the result
-			if (typeof tokenCount !== "number") {
-				console.warn("Cline <Language Model API>: Non-numeric token count received:", tokenCount)
-				return 0
-			}
-
-			if (tokenCount < 0) {
-				console.warn("Cline <Language Model API>: Negative token count received:", tokenCount)
-				return 0
-			}
-
-			return tokenCount
-		} catch (error) {
-			// Handle specific error types
-			if (error instanceof vscode.CancellationError) {
-				console.debug("Cline <Language Model API>: Token counting cancelled by user")
-				return 0
-			}
-
-			const errorMessage = error instanceof Error ? error.message : "Unknown error"
-			console.warn("Cline <Language Model API>: Token counting failed:", errorMessage)
-
-			// Log additional error details if available
-			if (error instanceof Error && error.stack) {
-				console.debug("Token counting error stack:", error.stack)
-			}
-
-			return 0 // Fallback to prevent stream interruption
-		}
+		// For other models, use an appropriate counting method because VSCode LM's countTokens is not reliable
+		// It gives incorrect counts for text parts, so we use an approx counting method
+		// This is necessary to avoid model hallucinations due to incorrect token counts
+		const textContent = typeof text === "string" ? text : this.extractTextFromMessage(text)
+		// Use 4 character-to-token ratio for Claude models
+		return Math.ceil(textContent.length / 3)
```

Comment:
I am doing 1/3 to over estimate the token usage so that we are less likely to hit token bounds ideally it should be 1/4.

NOTE: this is a really bad token counting method so perhaps I can use another approx token counting method.

Reply:
@abeatrix Can you send me a link to your approx token counts and I can use those instead if you prefer that.